src/lvalue.rs

```diff
@@ -214,7 +214,16 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                     _ => bug!("field access on non-product type: {:?}", base_layout),
                 };
 
-                let ptr = base_ptr.offset(offset.bytes());
+                let offset = match base_extra {
+                    LvalueExtra::Vtable(tab) => {
+                        let (_, align) = self.size_and_align_of_dst(base_ty, Value::ByValPair(PrimVal::Ptr(base_ptr), PrimVal::Ptr(tab)))?;
+                        // magical formula taken from rustc
+                        (offset.bytes() + (align - 1)) & (-(align as i64) as u64)
```

Comment:
This is `.align_to` on `Size`, isn't it? I mean, there shouldn't be more than one such function anywhere, in Rust (you could still have this generated as several instructions like we do in trans, but that's not relevant here